398 alignment with bfo

dr-shorthair opened the pull request:
Closes #398 

Preview https://raw.githack.com/w3c/sdw-sosa-ssn/398-alignment-with-bfo/ssn/index.html#BFO-alignment

dr-shorthair committed:
Initial alignments - incomplete and broken in a couple of places

dr-shorthair committed:
Merge branch 'gh-pages' into 398-alignment-with-bfo

dr-shorthair committed:
Documentation of BFO alignment

dr-shorthair committed:
Changelog entry; Format HTML

dr-shorthair committed:
bad heading

dr-shorthair committed:
Update alignments package diagram

dr-shorthair committed:
typos

dr-shorthair committed:
sosa:Procedure ~ prov:Plan subClassOf cco:Information Content Entity - per https://github.com/BFO-Mappings/PROV-to-BFO

dr-shorthair commented:
Alignment of `sosa:Procedure` now consistent with the alignment of `prov:Plan` to `CCO information content entity` as given in https://github.com/BFO-Mappings/PROV-to-BFO/blob/main/prov-cco-directmappings.ttl which is the basis of Nature paper https://doi.org/10.1038/s41597-025-04580-1 

I'm comfortable that this set of alignments is OK now. 
@alanruttenberg might have opinions.

dr-shorthair commented:
@ldesousa I checked to see if the CCO URIs dereference - I'm finding that they pull down TTL representations quite nicely 
e.g. https://www.commoncoreontologies.org/ont00000958 

The BFO ones go to Ontobee or OLS, except for http://purl.obolibrary.org/obo/BFO_0000057 (has participant) which is failing. 

I've reported it here https://github.com/BFO-ontology/BFO-2020/issues/154

alanruttenberg commented:
@dr-shorthair Did a first quick browse. I noticed Generation, Usage, and Invalidation are subclass of Instantaneous Event, which looked suspicious. 
In
```
Influence subClassOf:
(process or 'process boundary')
 and (not (process
 and 'process boundary'))
```
you don't need ```and (not (process and 'process boundary')``` because those are already disjoint.

When you write something like ```independent continuant or specifically dependent continuant or generically dependent continuant and not spatial region``` it allows for things that are not BFO entities, because the complement of spatial region includes owl:Things that are not BFO entities. Is there anything that isn't a BFO entity? If not the above would be better written as ```bfo:entity and not occurrent and not spatial region```

dr-shorthair commented:
Thanks @alanruttenberg . 

I think your comment relates to the PROV-BFO alignment from the Nature Scientific Data paper - Figure 2, where PROV Influence is shown. We aren't responsible for that work. I just used it to provide some perspectives, since we had already [aligned SSN with PROV in the 2017 edition of SSN](https://www.w3.org/TR/vocab-ssn/#PROV_Alignment). 

What we request now is a review of the **direct alignment of SSN to BFO/CCO** which can be inspected in draft [tabulated here](https://raw.githack.com/w3c/sdw-sosa-ssn/398-alignment-with-bfo/ssn/index.html#BFO-alignment) and in the [RDF (Turtle) representation here](https://github.com/w3c/sdw-sosa-ssn/blob/398-alignment-with-bfo/ssn/rdf/ontology/alignments/sosa-bfo.ttl)

dr-shorthair commented:
Please note that I have erred on the generalized side in the BFO/CCO hierarchy, and @alanruttenberg may suggest some tighter alignments for e.g. sosa:Sample.

dr-shorthair committed:
Use 'or' instead of &cup; for union classes
Use mouseover definitions for BFO/CCO

dr-shorthair committed:
move symbol explanation to Class and Property sections

dr-shorthair committed:
Remove local logic notation explanation

dr-shorthair committed:
BFO/CCO not DUL

dr-shorthair committed:
Merge branch 'gh-pages' into 398-alignment-with-bfo

# Conflicts:
#	ssn/chapters/Modularization.html
#	ssn/images/SSN-alignment-modules.svg

dr-shorthair committed:
Merge branch 'gh-pages' into 398-alignment-with-bfo

dr-shorthair committed:
Merge branch 'gh-pages' into 398-alignment-with-bfo

maximelefrancois86 reviewed (changes requested):
very minor changes:

in html: 

* order in section 4.3 is not the same as in section 7
* the alignment of Property to Quality forces Properties to be "foi specific" (see examples). I think we should open up the alignment to  **BFO19 (Quality) OR BFO31 (generically dependent continuent)**
* sosa`:implements` cannot be aligned to BFO34 (function), which is a class
* can't find any object property in the version of BFO I found. and BFO55 (realizes) and BFO57 (has participant) don't dereference
* change/define style for explanation span next to rhs ? (monospaced font not justified, as the text is not part of the concept CURIE)

dr-shorthair committed:
Responding to comments by @max

dr-shorthair committed:
format BFO labels

dr-shorthair committed:
formatting

dr-shorthair committed:
xref to alignment notation

dr-shorthair committed:
Merge branch 'gh-pages' into 398-alignment-with-bfo

dr-shorthair commented:
Thanks @maximelefrancois86 - I've fixed all of these _except_ the alignment of `sosa:implements` - see https://raw.githack.com/w3c/sdw-sosa-ssn/398-alignment-with-bfo/ssn/index.html#BFO-alignment 

@alanruttenberg do you have any suggestions for this? `implements` links an instance of a `System` (i.e. a Sensor or Actuator or Sampler) to the description of the `Procedure` (i.e. the recipe or specification) that it follows. We have mapped `System` to `Independent Continuant` and `Procedure` to `Information Content Entity`. What are the object-property options that tie an independent continuant to a generically dependent continuant?

dr-shorthair committed:
Make implements subproperty of is subject of

dr-shorthair committed:
Merge branch 'gh-pages' into 398-alignment-with-bfo

# Conflicts:
#	ssn/index.html

dr-shorthair commented:
I have suggested alignment of `implements` to `is subject of` which is a kind of generic `independent continuant` --> `generically dependent continuant` relationship. 

I think this now clears all the concerns raised by @maximelefrancois86 

Preview https://raw.githack.com/w3c/sdw-sosa-ssn/398-alignment-with-bfo/ssn/index.html#BFO-alignment

dr-shorthair committed:
Merge branch 'gh-pages' into 398-alignment-with-bfo

dr-shorthair committed:
Add explanation for Platform, Procedure, Property alignments;
make Procedure alignment to Descriptive Information Content Entity 9rather than its more general superclass.

dr-shorthair committed:
Links instead of monospace for class names

dr-shorthair commented:
Please review this PR @maximelefrancois86 @kjano @ldesousa @oldskeptic

dr-shorthair committed:
Merge branch 'gh-pages' into 398-alignment-with-bfo

dr-shorthair commented:
@alanruttenberg @jjbittner I would be interested in your review of https://raw.githack.com/w3c/sdw-sosa-ssn/398-alignment-with-bfo/ssn/index.html#BFO-alignment

dr-shorthair commented:
@JJBittner have you been able to review this?

dr-shorthair commented:
If #416 is accepted (`Asset` - superclass of `Platform` and `System`), the BFO alignment must be supplemented to match. 

`sosa:Asset` &sube; `obo:BFO_0000004` 
                                # independent continuant

dr-shorthair committed:
Procedure is a _prescriptive_ information content entity (not _descriptive_)

dr-shorthair commented:
A specific question to @JJBittner and @alanruttenberg : 

I adjusted the alignment of `sosa:Procedure` to _prescriptive_ information content entity.  
Does this make sense to you? 

See https://raw.githack.com/w3c/sdw-sosa-ssn/398-alignment-with-bfo/ssn/index.html#BFO-alignment and https://github.com/w3c/sdw-sosa-ssn/blob/398-alignment-with-bfo/ssn/rdf/ontology/alignments/sosa-bfo.ttl

TallTed commented on ssn/chapters/BFO-alignment.html:
```diff
@@ -138,7 +137,8 @@ <h4>Class Alignments</h4>
   <li>
-    A <a>sosa:Procedure</a> is a re-usable recipe or protocol or system datasheet, thus it is purely information content, in particular it describes the steps that a system must implement.
+    A <a>sosa:Procedure</a> is a re-usable recipe or protocol or system datasheet, thus it is purely information content. 
+    In particular it describes the actions and settings that a system must implement.
```
```suggestion
    In particular, it describes the actions and settings that a system must implement.
```

dr-shorthair committed:
Add alignment for sosa:Stimulus

TallTed commented on ssn/chapters/BFO-alignment.html:
```diff
@@ -115,6 +116,15 @@ <h4>Class Alignments</h4>
+      <span class="symbol logic">⊑</span>
+      <span class="rhs">
+        <span
+          title="process: p is a process means p is an occurrent that has some temporal proper part and for some time t, p has some material entity as participant">obo:BFO_0000015</span>
```
```suggestion
          title="process: 'p is a process' means p is an occurrent that has some temporal proper part, and for some time, t, p has some material entity as participant">obo:BFO_0000015</span>
```

TallTed commented on ssn/chapters/BFO-alignment.html:
```diff
@@ -133,17 +143,23 @@ <h4>Class Alignments</h4>
-    A <a>sosa:Platform</a> may be a `material entity` such as a buoy, station, satellite, etc. 
-    Or it may just be a location without any physical infrastructure.  
+    A <a>sosa:Platform</a> may be a `material entity` such as a buoy, station, satellite, etc.
+    Or it may just be a location without any physical infrastructure.
```
```suggestion
    A <a>sosa:Platform</a> may be a `material entity` such as a buoy, station, satellite, etc.,
    or it may just be a location without any physical infrastructure.
```

TallTed commented on ssn/chapters/BFO-alignment.html:
```diff
@@ -133,17 +143,23 @@ <h4>Class Alignments</h4>
   </li>
   <li>
-    A <a>sosa:Procedure</a> is a re-usable recipe or protocol or system datasheet, thus it is purely information content. 
+    A <a>sosa:Procedure</a> is a re-usable recipe or protocol or system datasheet, thus it is purely information
```
```suggestion
    A <a>sosa:Procedure</a> is a re-usable recipe, protocol, or system datasheet; thus, it is purely information
```

TallTed commented on ssn/chapters/BFO-alignment.html:
```diff
@@ -133,17 +143,23 @@ <h4>Class Alignments</h4>
-    Or it may be specific to a particular individual. 
-    See <a href="#feature-specific-properties"></a> for further discussion. 
+    A <a>sosa:Property</a> may be generic, so that it applies to many entities.
+    Or it may be specific to a particular individual.
```
```suggestion
    A <a>sosa:Property</a> may be generic, so that it applies to many entities,
    or it may be specific to a particular individual.
```

alanruttenberg commented:
> A specific question to @JJBittner and @alanruttenberg :
> 
> I adjusted the alignment of `sosa:Procedure` to _prescriptive_ information content entity. Does this make sense to you?

Yes

alanruttenberg commented:
@dr-shorthair 

> `implements` links an instance of a `System` (i.e. a Sensor or Actuator or Sampler) to the description of the `Procedure` (i.e. the recipe or specification) that it follows. We have mapped `System` to `Independent Continuant` and `Procedure` to `Information Content Entity`. What are the object-property options that tie an independent continuant to a generically dependent continuant?

Tagging @APCox to check this, as I haven't exercised this part of CCO.

There's the class 
> *Performance Specification* A Directive Information Content Entity that prescribes some aspect of the behavior of a participant in a Process given one or more operating conditions.

It uses the relation *describes condition* to relate the Performance Specification to the participant (best I can tell).

>p describes_condition c iff p is an instance of a Performance Specification and c is an instance of Entity and p has part d, and d is an instance of a Descriptive Information Content Entity, and p prescribes an entity s, and d describes c, and c is an entity that is causally relevant to s existing as prescribed by p.

Which is a little hard to follow, but I think in this situation, p (is part) of the procedure, c is the system, s is the procedure executed.

If you want to go from system to procedure (the reverse direction) the inverse is the relation *condition described by* . So System *condition described by* Procedure?

dr-shorthair commented:
[prescribes](https://www.commoncoreontologies.org/ont00001942) (sibling of `describes condition`) and inverse [prescribed by](https://www.commoncoreontologies.org/ont00001920) (sibling of `condition described by`) look more relevant to me?

dr-shorthair committed:
Replace `is subject of` with `prescribed by` alignment with `implements`

dr-shorthair committed:
ditto for `usedProcedure`

dr-shorthair committed:
Merge branch 'gh-pages' into 398-alignment-with-bfo

dr-shorthair committed:
typo - Feature not Fature

dr-shorthair committed:
Align Execution (and subclasses) with "Planned Act" (and subclasses)

dr-shorthair committed:
alignment issue ...

dr-shorthair committed:
alignment issue ...

dr-shorthair committed:
Merge remote-tracking branch 'origin/398-alignment-with-bfo' into 398-alignment-with-bfo

alanruttenberg commented:
I am was confused by which file I should be looking at. The issues mentioned in the previous version of the comment do not occur this  [sosa-bfo.ttl](https://raw.githubusercontent.com/w3c/sdw-sosa-ssn/refs/heads/398-alignment-with-bfo/ssn/rdf/ontology/alignments/sosa-bfo.ttl)

[deleted]

dr-shorthair committed:
Add ranges for sosa:hasResult

dr-shorthair committed:
Deployment is a kind of planned-act
Stimulus ~ Cause

dr-shorthair committed:
Deployment is a kind of planned-act
Stimulus ~ Cause

dr-shorthair committed:
Merge remote-tracking branch 'origin/398-alignment-with-bfo' into 398-alignment-with-bfo

dr-shorthair commented:
@alanruttenberg yes, [sosa-bfo.ttl](https://raw.githubusercontent.com/w3c/sdw-sosa-ssn/refs/heads/398-alignment-with-bfo/ssn/rdf/ontology/alignments/sosa-bfo.ttl) is the correct file. 

I've made some refinements following some interactions with Danielle Limbaugh and J Bittner. 
`Procedure`, `Stimulus`, `hasResult` all significantly improved. 
I'm wondering if the mapping for `Stimulus` should be collapsed to only `Cause`. 

See https://raw.githack.com/w3c/sdw-sosa-ssn/398-alignment-with-bfo/ssn/index.html#BFO-alignment

alanruttenberg commented:
> I've made some refinements following some interactions with Danielle Limbaugh and J Bittner. `Procedure`, `Stimulus`, `hasResult` all significantly improved. I'm wondering if the mapping for `Stimulus` should be collapsed to only `Cause`.
> 
> See https://raw.githack.com/w3c/sdw-sosa-ssn/398-alignment-with-bfo/ssn/index.html#BFO-alignment

The change history is too compact for me to understand the reasoning behind the changes. If you've had substantive interactions with others (Limbaugh/Bittner) can you share these here so that I can take into account their reasoning? I'm not sure I will come to the same conclusions they have, but I can't tell if I don't know the substance of them.

dr-shorthair commented:
1. I had [already proposed to align `sosa:Procedure` to cco:ont00000965](https://github.com/w3c/sdw-sosa-ssn/pull/401#issuecomment-3770855304) `prescriptive information content entity`, which also led to [adjustment for `sosa:implements` and `sosa:usedProcedure` using cco:ont00001920 `prescribed by` for consistency](https://github.com/w3c/sdw-sosa-ssn/pull/401#issuecomment-3821812771) (see above). 

2. Limbaugh provided a spreadsheet for an alignment of the 2017 edition of SSN to BFO/CCO. Mostly that was consistent with the conversations that we had already had. But from that I learned about cco:ont00000978 `Cause` which is very relevant to `sosa:Stimulus`. 

3. Bittner drew my attention to cco:ont00000228 `Planned Act` which is clearly the superclass (or perhaps equivalent) of `sosa:Execution`. I then hunted through _its_ subclasses to find some matches for `sosa:Actuation`, `sosa:Observation`, and `sosa:Sampling`. 

4. `sosa:hasResult` has no global range constraint specified in SSN, but I proposed local constraints in the context of `sosa:Actuation`, `sosa:Observation`, and `sosa:Sampling` using BFO/CCO classes as the range. 

Remember that in general SSN has much weaker axiomatization than the BFO/CCO style, in particular no global domain/range constraints. But the SSN-BFO alignment file does provide a home for enhancing this.

dr-shorthair committed:
Stimulus subClassOf `Cause`

dr-shorthair committed:
Stimulus subClassOf `Cause`

dr-shorthair committed:
Merge remote-tracking branch 'origin/398-alignment-with-bfo' into 398-alignment-with-bfo

dr-shorthair committed:
Merge branch 'gh-pages' into 398-alignment-with-bfo

dr-shorthair committed:
Prescriptive renamed Descriptive in CCO

dr-shorthair commented:
@DanielleZLimbaugh here's the conversation on the SSN-BFO alignment

TallTed commented:
Error in commit message, [`Prescriptive renamed Descriptive in CCO`](https://github.com/w3c/sdw-sosa-ssn/pull/401/commits/5dfed4899b5ccd8122885b2bb572e962d70dafe2).

Should be `Prescriptive renamed Directive in CCO`. *VERY* different meaning, even if only cosmetic!

alanruttenberg commented:
> _VERY_ different meaning, even if only cosmetic!

How would you characterize the difference? In any case, it's the definition that governs, not the label. 

>An Information Content Entity that consists of a set of propositions or images (as in the case of a blueprint) that prescribe some Entity.

dr-shorthair commented:
It is the same class, I just fixed the label in our documentation to match the current label in CCO.

dr-shorthair committed:
Merge branch 'gh-pages' into 398-alignment-with-bfo

dr-shorthair committed:
Minor formatting

dr-shorthair committed:
Add Asset

dr-shorthair committed:
Merge branch '324-modspec-identifiers' into 398-alignment-with-bfo

# Conflicts:
#	ssn/chapters/Sampling.html

dr-shorthair committed:
minor styling

dr-shorthair committed:
Pescriptive not Directive

dr-shorthair committed:
Add Limbaugh to Contributors

dr-shorthair commented:
This is ready to go now. 
Limbaugh may add some more detail in due course, but the framework is sound.

ldesousa commented on ssn/chapters/BFO-alignment.html:
```diff
@@ -0,0 +1,306 @@
+    </li>
+    <li>
+      <span class="lhs"><a>sosa:Asset</a></span>
+      <span class="symbol logic">≡</span>
```
```suggestion
      <span class="symbol logic">⊑</span>
```

ldesousa commented:
Just a heads up, there seem to be commits from #421 within this PR. Might result in conflicts later.

ldesousa reviewed (dismissed):
Please check the sign here: https://github.com/w3c/sdw-sosa-ssn/pull/401#discussion_r2924767437

oldskeptic reviewed (approved):
Looks good to me.

dr-shorthair committed:
Update ssn/chapters/BFO-alignment.html

Co-authored-by: Luís de Sousa <luis.de.sousa@protonmail.ch>

dr-shorthair committed:
Merge remote-tracking branch 'origin/gh-pages' into 398-alignment-with-bfo

# Conflicts:
#	ssn/config.js
#	ssn/index.html
#	ssn/style.css

dr-shorthair committed:
Asset subclass not equivelent of independent continuant

dr-shorthair committed:
Merge remote-tracking branch 'origin/398-alignment-with-bfo' into 398-alignment-with-bfo

dr-shorthair committed:
Clarify alignments related to top level BFO classes
